fe/fe-core/src/main/java/org/apache/doris/planner/external/LoadPlanner.java

```diff
@@ -0,0 +1,194 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.planner.external;
+
+import org.apache.doris.analysis.Analyzer;
+import org.apache.doris.analysis.BrokerDesc;
+import org.apache.doris.analysis.DataDescription;
+import org.apache.doris.analysis.DescriptorTable;
+import org.apache.doris.analysis.SlotDescriptor;
+import org.apache.doris.analysis.TupleDescriptor;
+import org.apache.doris.analysis.UserIdentity;
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.catalog.Type;
+import org.apache.doris.common.Config;
+import org.apache.doris.common.UserException;
+import org.apache.doris.load.BrokerFileGroup;
+import org.apache.doris.planner.DataPartition;
+import org.apache.doris.planner.FileLoadScanNode;
+import org.apache.doris.planner.OlapTableSink;
+import org.apache.doris.planner.PlanFragment;
+import org.apache.doris.planner.PlanFragmentId;
+import org.apache.doris.planner.PlanNodeId;
+import org.apache.doris.planner.ScanNode;
+import org.apache.doris.task.LoadTaskInfo;
+import org.apache.doris.thrift.TBrokerFileStatus;
+import org.apache.doris.thrift.TExecPlanFragmentParams;
+import org.apache.doris.thrift.TFileType;
+import org.apache.doris.thrift.TUniqueId;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+public abstract class LoadPlanner {
+
+    private static final Logger LOG = LogManager.getLogger(LoadPlanner.class);
+
+    public List<List<TBrokerFileStatus>> fileStatusesList;
+
+    public int filesAdded;
+
+    protected LoadTaskInfo taskInfo;
+
+    protected OlapTable table;
+
+    // Something useful
+    // ConnectContext here is just a dummy object to avoid some NPE problem, like ctx.getDatabase()
+    protected Analyzer analyzer;
+
+    protected DescriptorTable descTable;
+
+    public void preparePlan(List<List<TBrokerFileStatus>> fileStatusesList, int filesAdded) {
```

Comment:
If we use insert into select to replace current stream load logic, this planner maybe useless....

Reply:
I don't quite understand the meaning of your comment in this line. I think the streamloadplanner won't call the prepareplan method, it's just abstracted out to unify the plan method.